openmc/mgxs/mgxs.py

```diff
@@ -5891,3 +5897,466 @@ def get_units(self, xs_type='macro'):
         else:
             raise ValueError('Unable to return the units of InverseVelocity'
                              ' for xs_type other than "macro"')
+
+class SurfaceMGXS(MGXS,metaclass=ABCMeta):
+    """An abstract multi-group cross section for some energy group structure
+    on the surfaces of a mesh domain.
+    This class can be used for both OpenMC input generation and tally data
+    post-processing to compute surface- and energy-integrated multi-group cross
+    section for multi-group neutronics calculations.
+    NOTE: Users should instantiate the subclasses of this abstract class.
+    Parameters
+    ----------
+    domain : openmc.RegularMesh
+        The domain for spatial homogenization
+    domain_type : {'mesh'}
+        The domain type for spatial homogenization
+    energy_groups : openmc.mgxs.EnergyGroups
+        The energy group structure for energy condensation
+    by_nuclide : bool
+        If true, computes cross sections for each nuclide in domain
```

Comment:
You're right, in fact an error occurs if by_nuclide = True, since the domain is a mesh, and a mesh doesn't have nuclides. 
I will change it so that by_nuclide is inherited from MGXS, but raises an explicit error if it's True. Then I can eliminate it as a parameter, attribute, and in all the if statements of the SurfaceMGXS class and Current class.

Reply:
Good point.
I don't actually know whether an attribute that is both unused and should not be used should be present in the docstrings. @paulromano or @nelsonag  do you have an opinion on this ?